### DISCUSSION TITLE
Encountering New Error: "Error: No matching key. history: 1686608831790513"

### DISCUSSION THREAD

**Discussion by andrewdrossi** [WalletConnect/walletconnect-monorepo#2612] [Developer Support] [answered]
This may be a tough one.

Context:
I am using WalletConnect V2 EthereumProvider and having users connect their wallet via the QR code. I then convert the provider into a Web3Provider using ethers.providers.Web3Provider(). I am using both web3.js and ethers.js as a stop gap to get V2 working while we migrate fully from web3.js to ethers.js since V2 does not work with web3.js by default. In the meantime, ethers essentially is only used to deal with interacting with EthereumProvider for V2. This provider (used with Web3js) works for transaction creation, submission, and for provider events. 

Occasionally, when I connect using Rainbow, I get the following error:

**_Uncaught (in promise) Error: No matching key. history: 1686608831790513_**

I am trying to isolate the issue but am having trouble. Everything is working as expected in my application: QR code modal pops up, on scan Rainbow prompts to connect, and I am able to connect and use the provider for interacting with smart contracts as well as provider based events. However, I am seeing this error in my chrome console and want to address it.

```
// Instantiate WC provider

let wcv2 = await EthereumProvider.init({
                projectId: process.env.REACT_APP_WALLET_CONNECT_PROJECT_ID,
                chains: [1]
})
ethersProvider = new ethers.providers.Web3Provider(wcv2)

// Open QR Code Modal & Connect

const address = await ethersProvider.provider.enable()
return Web3.utils.toChecksumAddress(address[0])
```

**Comment by Cali93** [comment, marked as answer]
Hey @andrewdrossi,

Which version of EthereumProvider are you using ? 
Can you try updating to latest and let us know if the error still occurs ?

**Comment by andrewdrossi** [comment]
Hello,

I was using 2.4.4 but I have since upgraded to 2.8.0 and it seems to be
working.
